libcni/api.go

```diff
@@ -391,6 +406,65 @@ func (c *CNIConfig) GetNetworkCachedConfig(net *NetworkConfig, rt *RuntimeConf)
 	return c.getCachedConfig(net.Network.Name, rt)
 }
 
+// GetCachedAttachments returns a list of network attachments from the cache.
+// The returned list will be filtered by the containerID if the value is not empty.
+func (c *CNIConfig) GetCachedAttachments(containerID string) ([]*NetworkAttachment, error) {
+	dirPath := filepath.Join(c.getCacheDir(&RuntimeConf{}), "results")
+	entries, err := os.ReadDir(dirPath)
+	if err != nil {
+		return nil, err
+	}
+
+	fileNames := make([]string, 0, len(entries))
+	for _, e := range entries {
+		fileNames = append(fileNames, e.Name())
+	}
+	sort.Strings(fileNames)
```

Comment:
Why sort?

Reply:
Good question. I didn't author this code but ported from [ocicni](https://github.com/cri-o/ocicni/blob/master/pkg/ocicni/ocicni.go#L592). And I had same question when reading the code.

I guess the intention was making it easier for the caller to process the results in logical groups, because the cache file names are in the format of "network-container-ifname" as defined [here](https://github.com/containernetworking/cni/blob/6626820a52c9d460651c805d81f6307cf2c4b865/libcni/api.go#L223).